### PR TITLE
Merge https://github.com/kubernetes-sigs/cluster-api-operator:main (e6bc25c) into main

### PR DESCRIPTION
**Pull request by damdo**
This PR merges upstream's https://github.com/kubernetes-sigs/cluster-api-operator:main (e6bc25c) into `main`.

Check [this comment](https://github.com/openshift/cluster-capi-operator/pull/101#issue-1564195291) for testing info.

### CODE REVIEW

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.19-openshift-4.13
+  tag: rhel-8-release-golang-1.17-openshift-4.11
```
Comment (elmiko) [inline .ci-operator.yaml:4]: it looks like we are undoing a recent ART commit with this

File: Dockerfile [hunk 2]
```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 
 # Build the manager binary
-FROM golang:1.17.0 as builder
+FROM golang:1.19.0 as builder
```
Comment (elmiko) [inline Dockerfile:18]: same here as well

**Review by elmiko — COMMENTED** [review]
this generally looks ok from the commit history and tests, but it seems like we are stomping on the golang and ocp versions.

File: Dockerfile (hunk 2 above)
Comment (damdo) [reply Dockerfile:18]: This one doesn't look to ever be updated by ART, it's the upstream one. We only use `openshift/Dockerfile.rhel`.

File: .ci-operator.yaml (hunk 1 above)
Comment (damdo) [reply .ci-operator.yaml:4]: Good catch! Weird it kept only the latest [UPSTREAM: <carry>: Updating ose-cluster-kube-cluster-api-operator ima…](https://github.com/openshift/cluster-api-operator/commit/c23bc67c8dd546ad6ed28d19a07b33fc9e79d54a) commit, but not the one prior, which updated this file to 1.19. Will bump it manually.

File: Dockerfile (hunk 2 above)
Comment (elmiko) [reply Dockerfile:18]: oops, yeah, i misread that one.

File: .ci-operator.yaml (hunk 1 above)
Comment (damdo) [reply .ci-operator.yaml]: Bumped

**Review by elmiko — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**